tests/components/monzo/test_config_flow.py

```diff
@@ -136,3 +141,155 @@ async def test_config_non_unique_profile(
+            "access_token": "mock-access-token",
+            "type": "Bearer",
+            "expires_in": 60,
+            "user_id": str(USER_ID),
+        },
+    )
+    with patch(
+        "homeassistant.config_entries.ConfigEntries.async_update_entry",
+        return_value=True,
+    ) as mock_setup:
+        result = await hass.config_entries.flow.async_configure(result["flow_id"])
+
+        mock_setup.assert_not_called()
+        assert result["type"] is FlowResultType.FORM
+        assert result["step_id"] == "await_approval_confirmation"
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={"confirm": True}
+    )
+
+    assert len(hass.config_entries.async_entries(DOMAIN)) == 1
+    assert result
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "reauth_successful"
+
+
+async def test_config_reauth_wrong_account(
+    hass: HomeAssistant,
+    hass_client_no_auth: ClientSessionGenerator,
+    aioclient_mock: AiohttpClientMocker,
+    polling_config_entry: MockConfigEntry,
+    current_request_with_host: None,
+) -> None:
+    """Test reauth with wrong account."""
+    await setup_integration(hass, polling_config_entry)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={
+            "source": SOURCE_REAUTH,
+            "entry_id": polling_config_entry.entry_id,
+        },
+        data=polling_config_entry.data,
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "reauth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+    state = config_entry_oauth2_flow._encode_jwt(
+        hass,
+        {
+            "flow_id": result["flow_id"],
+            "redirect_uri": "https://example.com/auth/external/callback",
+        },
+    )
+    assert result["url"] == (
+        f"{OAUTH2_AUTHORIZE}/?"
+        f"response_type=code&client_id={CLIENT_ID}&"
+        "redirect_uri=https://example.com/auth/external/callback&"
+        f"state={state}"
+    )
+    client = await hass_client_no_auth()
+    resp = await client.get(f"/auth/external/callback?code=abcd&state={state}")
+    assert resp.status == 200
+    assert resp.headers["content-type"] == "text/html; charset=utf-8"
+
+    aioclient_mock.clear_requests()
+    aioclient_mock.post(
+        OAUTH2_TOKEN,
+        json={
+            "refresh_token": "mock-refresh-token",
+            "access_token": "mock-access-token",
+            "type": "Bearer",
+            "expires_in": 60,
+            "user_id": 12346,
+        },
+    )
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"])
+    assert result
+    assert result["type"] == FlowResultType.ABORT
+    assert result["reason"] == "wrong_account"
+
+
+@patch(
+    "homeassistant.components.monzo.config_flow.MonzoFlowHandler.async_step_reauth",
+    wraps=MonzoFlowHandler.async_step_reauth,
+)
```

Comment:
What does this do?

Reply:
I mean, why do we need to check if its called? we can test the result of everything

Reply:
This was to keep the scope of the test small and specific - it doesn't really care if the reauth succeeds, only that the AuthorisationExpiredError triggers it

Reply:
you can then just check if there is a new flow created and that the config entry state is set to SETUP_ERROR

Reply:
Sorry, how do I check that without a patch or mock?

Reply:
You can setup the integration, and when the client is mocked to raise an auth error, you can check the state of the config entry

Reply:
It looks like the state never actually gets changed. It's not changed by the [DataUpdateCoordinator](https://github.com/home-assistant/core/blob/cc60fc6d9ff7435995936014ed60b3195794c35b/homeassistant/helpers/update_coordinator.py#L363C1-L378C64), [`start_reauth`](https://github.com/home-assistant/core/blob/cc60fc6d9ff7435995936014ed60b3195794c35b/homeassistant/config_entries.py#L992-L1008) or [`init_reauth`](https://github.com/home-assistant/core/blob/cc60fc6d9ff7435995936014ed60b3195794c35b/homeassistant/config_entries.py#L1010-L1049). Am I missing something?

Reply:
I think you can find this test in some test_init.py instead of the test_config_flow.py,